pyrightconfig.json

```diff
@@ -15,15 +17,13 @@
 	"typeCheckingMode": "basic",
 	// Too many issues caused by dynamic patching, still worth fixing when we can
 	"reportAttributeAccessIssue": "warning",
-	// Fails on Python 3.12 due to missing distutils and on cygwin CI tests
-	"reportAssignmentType": "warning",
+	// Fails on Python 3.12 due to missing distutils
 	"reportMissingImports": "warning",
-	"reportOptionalCall": "warning",
 	// FIXME: A handful of reportOperatorIssue spread throughout the codebase
 	"reportOperatorIssue": "warning",
 	// Deferred initialization (initialize_options/finalize_options) causes many "potentially None" issues
-	// TODO: Fix with type-guards or by changing how it's initialized
-	"reportArgumentType": "warning", // A lot of these are caused by jaraco.path.build's spec argument not being a Mapping https://github.com/jaraco/jaraco.path/pull/3
```

Comment:
This has been fixed upstream, so the comment doesn't apply anymore and those `reportArgumentType` left are of setuptool's own making.